### PR TITLE
MueLu: Refactor of Driver to allow for code reuse

### PR DESCRIPTION
**Pull request by csiefer2**
Auto-PR for SHA 1b1f315

### CODE REVIEW

**Comment by csiefer2** [comment]
Working on a multi-run example, using some of @aprokop 's work.

Also need to add some error handling (for when MueLu or Belos crash)

File: packages/muelu/test/scaling/Driver.cpp [hunk 1]
```diff
@@ -438,9 +438,9 @@ int main_(Teuchos::CommandLineProcessor &clp, Xpetra::UnderlyingLib& lib, int ar
         userParamList.set("Node Comm",nodeComm);
       }
 #endif
-      out2<<"*********** ParameterList ***********"<<std::endl;
+      out2<<"*********** MueLu ParameterList ***********"<<std::endl;
       out2<<mueluList;
-      out<<"*************************************"<<std::endl;
+      out<<"********************************************"<<std::endl;
```
Comment (jhux2) [inline packages/muelu/test/scaling/Driver.cpp]: @csiefer2 Should this be `out2`?

**Comment by csiefer2** [comment]
#$%^%^ing intel compiler.

File: packages/muelu/test/scaling/Driver.cpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/test/scaling/Driver.cpp]: Fixed

**Comment by jhux2** [comment]
> #$%^%^ing intel compiler.

🐙

**Comment by csiefer2** [comment]
Care to rebless, @jhux2 ?

**Pull request merged**